streamflow/data/remotepath.py

```diff
@@ -202,6 +204,83 @@ async def follow_symlink(
         return result.strip()
 
 
+async def get_mount_point(
+    context: StreamFlowContext,
+    connector: Connector,
+    location: AvailableLocation | None,
+    path: str,
+) -> str:
+    """
+    Get the mount point of a path in the location
+
+    :param context: the `StreamFlowContext` object with global application status.
+    :param connector: the `Connector` object to communicate with the location
+    :param location: the `ExecutionLocation` object with the location information
+    :param path: the path of which discover its mount point
+    :return: the mount point path
+    """
+    try:
+        return location.hardware.get_mount_point(path)
+    except KeyError:
+        mount_point = path
+        while (
+            mount_point := await follow_symlink(
+                context, connector, location.location, mount_point
+            )
+        ) is None:
+            mount_point = Path(mount_point).parent
+        # follow symlynk can return empty string
+        if not mount_point:
+            raise WorkflowExecutionException(
+                f"Impossible to find the mountpoint: the {path} path does not exist"
+            )
+        # todo: wraps not considered
+        location_mount_points = location.hardware.get_mount_points()
+        while (
+            str(mount_point) != os.sep and str(mount_point) not in location_mount_points
+        ):
+            mount_point = Path(mount_point).parent
+        location.hardware.get_storage(str(mount_point)).add_path(path)
+        return str(mount_point)
+
+
+async def get_storage_usages(
+    connector: Connector, location: ExecutionLocation, hardware: Hardware
+) -> MutableMapping[str, int]:
+    """
+    Get the actual size of the hardware storage paths
+
+    Warn. Storage keys are not mount points.
+    Each `HardwareRequirement` implementation can use the storage dictionary keys as it wants.
+    No assumption about the key meaning can be done.
```

Comment:
```suggestion
    Since the meaning of storage dictionary keys depends on each `HardwareRequirement` implementation, no assumption about the key meaning should be made.
```